### PR TITLE
Added MiDaS v2.1

### PR DESCRIPTION
**Pull request by AlexeyAB**
Update MiDaS v2 to v2.1: https://github.com/intel-isl/MiDaS#accuracy

* Large model: +10% accuracy
* Small model: 50x times faster than previous small model on Mobile Android/iOS (with about the same accuracy as the previous small model from the paper), it is real-time for Mobile devices

### CODE REVIEW

**Comment by maringeo** [comment]
By opening this pull request, you are agreeing to the TensorFlow Hub Terms of Service at www.tfhub.dev/terms, and your pull request is being reviewed.

File: tfhub_dev/assets/intel/midas/v2_1/2/lite/1.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# Lite intel/midas/v2_1/2
+A deployment format of Lite intel/midas/v2_1/2
+
+<!-- asset-path: https://github.com/intel-isl/MiDaS/releases/download/v2_1/model_opt.tflite -->
+<!-- module-type: image-depth-estimation -->
+<!-- parent-model: agripredict/disease-classification/1 -->
```
Comment (maringeo) [inline tfhub_dev/assets/intel/midas/v2_1/2/lite/1.md:6]: Is this the correct? Did you meant `intel/midas/v2_1/2`?

File: tfhub_dev/assets/intel/midas/v2_1/2/lite/1.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Lite intel/midas/v2_1/2
```
Comment (maringeo) [inline tfhub_dev/assets/intel/midas/v2_1/2/lite/1.md:1]: This should follow the direcotry structure, so it should be `intel/midas/v2_1/2/lite/1`

File: tfhub_dev/assets/intel/intel.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Publisher intel
```
Comment (maringeo) [inline tfhub_dev/assets/intel/intel.md:1]: Can you update the file location of all files in this PR to be under `assets/docs`?

File: tfhub_dev/assets/intel/midas/v2_1/1/1.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
+# Module intel/midas/v2_1/1
```
Comment (maringeo) [inline tfhub_dev/assets/intel/midas/v2_1/1/1.md:1]: In the current state of the PR, `intel/midas/v2_1/1` and `intel/midas/v2_1/2` represent two versions of the same model. There might be a more appropriate TF Hub concept we can use to represent the relationship between these models. What is the difference between `intel/midas/v2_1/1` and `intel/midas/v2_1/2`.

**Comment by AlexeyAB** [comment]
@maringeo Thanks, I fixed it.

> In the current state of the PR, intel/midas/v2_1/1 and intel/midas/v2_1/2 represent two versions of the same model. There might be a more appropriate TF Hub concept we can use to represent the relationship between these models. What is the difference between intel/midas/v2_1/1 and intel/midas/v2_1/2.

* intel/midas/v2_1/1 - is a large model with high accuracy
* intel/midas/v2_1/2 - is a small model with high speed (30 FPS on iPhone)

**Comment by maringeo** [comment]
> * intel/midas/v2_1/1 - is a large model with high accuracy
> * intel/midas/v2_1/2 - is a small model with high speed (30 FPS on iPhone)

Thanks @AlexeyAB, in that case these are two separate models and require different handles (handles are the unique idenitifiers for a model, namely `intel/midas/v2_1/1` and `intel/midas/v2_1/2`). If two handles differ only at the last number they would be intepreted as different versions of the same model (the last number in a handle is the version of the model).

One common naming pattern used in tfhub.dev is to append `_large` suffix to the large model name and `_small` suffix to the small model name. Of course this doesn't allow for a lot of flexibility if a third variant of the model comes up. So if you can think of a more meaninful model property to append to the name it would be great.

Sorry, one follow-up question: how do these two models relate to this one
https://github.com/tensorflow/tfhub.dev/blob/master/assets/docs/intel/midas/v2/1/1.md

**Comment by AlexeyAB** [comment]
@maringeo 
Should I name it:
`intel/midas/v2_1/1_large` and `intel/midas/v2_1/1_small` ?

> Sorry, one follow-up question: how do these two models relate to this one
> https://github.com/tensorflow/tfhub.dev/blob/master/assets/docs/intel/midas/v2/1/1.md

This is the old large model

In this PR:
* New large model - the same structure, but new weights: +10% accuracy
* Small model - different structure and weights.

**Comment by maringeo** [comment]
Thanks for the quick reply @AlexeyAB, I think the PR looks good now. Before merging, I've outlined a few refactorings that are required because of how the TF Hub model publishing pipeline is structured.

If the new large model has the same structure and provides better accuracy I think it should be published as a second version to the already existing assets/docs/intel/midas/v2/1/1.md. This will group the two models in the UI and will allow users to quickly discover the newer version of the model.

In order to add the new large model as second version of assets/docs/intel/midas/v2/1/1.md we should:

- rename the file from `1.md` to `2.md`
- move the file in the assets/docs/intel/midas/v2/1/ directory
- change the first line of `2.md` to be `# Module intel/midas/v2/2` and also update the `parent-model` in the Lite .md file
- optionally describe what changed between version 1 and version 2 in the `2.md` documentation(https://tfhub.dev/google/universal-sentence-encoder-large/5 has a nice changelog example)

Regarding the name/handle of the small model - I think `intel/midas/v2_1_small/1` or `intel/midas_small/v2_1/1` are both meaningful and will work (`intel/midas/v2_1/1_small` won't work since the last part of the handle is the version and it should be an integer and in this example it's `1_small`). Optionally, you can update the directory path to match the model handle.

Let me know if you run into any problems or if you have any questions.

**Comment by AlexeyAB** [comment]
@maringeo Thanks. Please check if everything is correct now.

**Comment by maringeo** [comment]
Thanks @AlexeyAB, everything looks good. I'll merge the PR and I'll add a comment on this thread once the models are live on https://tfhub.dev/.

**Comment by maringeo** [comment]
The models are live at:
https://tfhub.dev/intel/midas/v2_1_small/1
https://tfhub.dev/intel/lite-model/midas/v2_1_small/1/lite/1
https://tfhub.dev/intel/midas/v2/2

**Pull request merged**